Code/Tools/SceneAPI/SceneCore/Events/ExportProductList.cpp

```diff
@@ -105,12 +105,20 @@ namespace AZ
                 AZ_Assert(!id.IsNull(), "Provided guid is not valid");
                 AZ_Assert(!lod.has_value() || lod < 16, "Lod value has to be between 0 and 15 or disabled.");
 
-                size_t index = m_products.size();
-                m_products.emplace_back(AZStd::move(filename), id, assetType, lod, subId, dependencyFlags);
-                return m_products[index];
+                ExportProduct exportProduct(AZStd::move(filename), id, assetType, lod, subId, dependencyFlags);
+                auto iter = m_products.find(exportProduct);
+                if (iter != m_products.end())
+                {
+                    return *iter;
+                }
+                else
+                {
+                    auto insertion = m_products.insert(AZStd::move(exportProduct));
+                    return *insertion.first;
+                }
```

Comment:
Suggestion: you could simplify this by just returning the results of the insertion.
```suggestion
                // Either insert the new export product if it's the first time this product is being produced,
                // or else silently fail the insert if it's a duplicate and return the previously-inserted product.
                return m_products.insert(ExportProduct(AZStd::move(filename), id, assetType, lod, subId, dependencyFlags).first;
```